src/lib/hidden_node.ts

```diff
@@ -0,0 +1,31 @@
+import {Context} from './context';
+
+/**
+ * Condition for applying rule 2A
+ * @param {HTMLElement} elem - The element whose accessible name is being calculated
+ * @param {Context} context - Additional information relevant to the name computation for elem
+ * @return {boolean} - Whether of not elem satisfies the condition for rule 2A
+ */
+function hiddenNodeCondition(elem: HTMLElement, context: Context): boolean {
+  const hidden: boolean = elem.hasAttribute('hidden');
```

Comment:
'hidden' is actually not well defined in the acc-name document, but I am pretty sure it is not just based on the 'hidden' attribute.

I think you will also need to check aria-hidden="true" & potentially CSS-based ways of hiding things. Typically having display: none also causes a node to be considered "hidden". I am not sure if there is a place that defines this precisely.

Reply:
also note that hidden applies to an entire subtree, so the node is considered hidden if any of its parents are hidden.